pkg/appmanager/appManager_test.go

```diff
@@ -3443,6 +3443,176 @@ var _ = Describe("AppManager Tests", func() {
 					flatDg = nsMap.FlattenNamespaces()
 					Expect(flatDg).To(BeNil(), "should not have data")
 				})
+
+				It("manages alternate backends for routes (pool creation)", func() {
+					svc1Name := "foo"
+					svc2Name := "bar"
+					svc3Name := "baz"
+					fooSvc := test.NewService(svc1Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37001}})
+					mockMgr.addService(fooSvc)
+					barSvc := test.NewService(svc2Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37002}})
+					mockMgr.addService(barSvc)
+					bazSvc := test.NewService(svc3Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37003}})
+					mockMgr.addService(bazSvc)
+
+					spec := routeapi.RouteSpec{
+						Host: "foo.com",
+						Path: "/foo",
+						To: routeapi.RouteTargetReference{
+							Kind: "Service",
+							Name: "fake1", // start with non-existent service
+						},
+						AlternateBackends: []routeapi.RouteTargetReference{
```

Comment:
Just a FYI - If the version of openshift supports A/B, it will default to a weight of 100 for any service that does not specify a weight (including the 'To' service).  If openshift doesn't support A/B, the weight will be ignored (and you will never have a alternateBackends field.